frame/staking/src/lib.rs

```diff
@@ -738,6 +738,13 @@ pub trait Config: frame_system::Config + SendTransactionTypes<Call<Self>> {
 		DataProvider = Module<Self>,
 	>;
 
+	/// Something that provides the election functionality at genesis.
+	type GenesisElectionProvider: frame_election_provider_support::ElectionProvider<
+		Self::AccountId,
+		Self::BlockNumber,
+		DataProvider = Module<Self>,
+	>;
```

Comment:
I was thinking instead of having an additional associated type we could have an additional field in the genesis config: `genesis_era: Vec<(AccountId, Exposure)>`.
Dev chain can simply use the the default exposure, and real chain can put a sensible value.

But your solution is good to me

Reply:
We still need a way to divert the call path of `new_session` on genesis from the rest, so I think there's still going to be the need for some of this code unfortunately.

Reply:
if there is a genesis_era, then it would work as long as new_session doesn't trigger a new era.

But yes your solution is cleaner IMO